crates/ide-assists/src/handlers/promote_local_to_const.rs

```diff
@@ -75,24 +72,28 @@ pub(crate) fn promote_local_to_const(acc: &mut Assists, ctx: &AssistContext<'_>)
         AssistId("promote_local_to_const", AssistKind::Refactor),
         "Promote local to constant",
         target,
-        |builder| {
+        |edit| {
             let name = to_upper_snake_case(&name.to_string());
             let usages = Definition::Local(local).usages(&ctx.sema).all();
             if let Some(usages) = usages.references.get(&ctx.file_id()) {
                 for usage in usages {
-                    builder.replace(usage.range, &name);
+                    let Some(usage) = usage.name.as_name_ref().cloned() else {
+                        continue
+                    };
+                    let usage = edit.make_mut(usage);
+                    ted::replace(usage.syntax(), make::name_ref(&name).clone_for_update().syntax());
```

Comment:
Haven't checked if this is possible, but can we hoist `make::name_ref().clone_for_update()` up outside the loop?

Reply:
Only the `make::name_ref()` portion. We need a fresh `clone_for_update()` copy for each usage we want to replace, otherwise all of the usages share the same mutable syntax tree and only one gets an actual parent.

Added a test to check that it doesn't panic since all of the other tests.

Reply:
Ah right, understood. Thanks a bunch for checking and adding a test for this!